common/scriptcore/api_game_methods.cpp

```diff
@@ -40,209 +43,86 @@
 
 #include "api_game_methods.h"
 
-/**
-   Return the current turn.
- */
-int api_methods_game_turn(lua_State *L)
-{
-  LUASCRIPT_CHECK_STATE(L, 0);
-
-  return game.info.turn;
-}
-
-/**
-   Return the current year.
- */
-int api_methods_game_year(lua_State *L)
-{
-  LUASCRIPT_CHECK_STATE(L, 0);
+namespace sol {
 
-  return game.info.year;
-}
-
-/**
-   Return the current year fragment.
- */
-int api_methods_game_year_fragment(lua_State *L)
-{
-  LUASCRIPT_CHECK_STATE(L, 0);
-
-  return game.info.fragment_count;
-}
-
-/**
-   Return the current year fragment.
- */
-const char *api_methods_game_year_text(lua_State *L)
-{
-  LUASCRIPT_CHECK_STATE(L, nullptr);
-
-  return calendar_text();
-}
-
-/**
-   Return the current turn, as if real turns started from 0.
- */
-int api_methods_game_turn_deprecated(lua_State *L)
-{
-  LUASCRIPT_CHECK_STATE(L, 0);
-
-  qCWarning(deprecations_category,
-            "Deprecated: lua construct \"game:turn\", deprecated "
-            "since \"3.0\", used. "
-            "Use \"game:current_turn\" instead.");
-
-  if (game.info.turn > 0) {
-    return game.info.turn - 1;
+namespace stack {
+template <> struct unqualified_pusher<::QString> {
+  static int push(lua_State *L, const ::QString &str)
+  {
+    return stack::push(L, qUtf8Printable(str));
   }
+};
 
-  return game.info.turn;
-}
-
-/**
-   Return name of the current ruleset.
- */
-const char *api_methods_game_rulesetdir(lua_State *L)
-{
-  Q_UNUSED(L)
-  return game.server.rulesetdir;
-}
-
-/**
-   Return name of the current ruleset.
- */
-const char *api_methods_game_ruleset_name(lua_State *L)
-{
-  Q_UNUSED(L)
-  return game.control.name;
-}
-
-/**
-   Return TRUE if pbuilding is a wonder.
- */
-bool api_methods_building_type_is_wonder(lua_State *L,
-                                         Building_Type *pbuilding)
-{
-  LUASCRIPT_CHECK_STATE(L, false);
-  LUASCRIPT_CHECK_SELF(L, pbuilding, false);
+template <> struct unqualified_getter<::QString> {
+  static ::QString get(lua_State *L, int index)
+  {
+    return QString(stack::get<const char *>(L, index));
+  }
+};
```

Comment:
I guess this should go to a header and then we might be able to drop the `toStdString` in the code calling signals.